Algorithmia/Algorithm/Algorithm.swift

```diff
@@ -16,14 +16,40 @@ class Algorithm {
         self.algoRef = algoRef
     }
     
-    func pipe(input:AnyObject!, completion:@escaping AlgoCompletionHandler) -> AlgoRequest? {
+    func pipe(input:Any!, completion:@escaping AlgoCompletionHandler) -> AlgoRequest? {
         if let stringInput = input as? String {
-            return client?.apiClient.post(path: algoRef.getPath(), data: AlgoStringEntity(entity: stringInput), completion: completion )
+            return pipe(text: stringInput, completion: completion)
         }
-        return nil
+        else {
+            return pipe(json: input, completion: completion)
+        }
+    }
+    
+    @discardableResult func pipe(text:String!, completion:@escaping AlgoCompletionHandler) -> AlgoRequest? {
+        return client?.apiClient.post(path: algoRef.getPath(), data: AlgoStringEntity(entity: text), completion: completion)
+        
     }
-    func pipe(input:String!, completion:@escaping AlgoCompletionHandler) -> AlgoRequest? {
-        return client?.apiClient.post(path: algoRef.getPath(), data: AlgoStringEntity(entity: input), completion: completion)
+    
+    @discardableResult func pipe(json:Any!, completion:@escaping AlgoCompletionHandler) -> AlgoRequest? {
+        do {
+            let entity = try AlgoJSONEntity(entity: json)
+            return client?.apiClient.post(path: algoRef.getPath(), data: entity, completion: completion)
+        } catch{
+            completion(AlgoResponse(),AlgoError.DataError("Data can not be serialized"))
+            return nil
+        }
         
+        
+        
+    }
+    
+    @discardableResult func pipe(json:String!, completion:@escaping AlgoCompletionHandler) -> AlgoRequest? {
```

Comment:
I had to chew on this a bit, but i think it's more fitting to use a named parameter like `rawJson:String!` for this case because it's technically possible to serialize a regular string (e.g. `"foo"`) as JSON (e.g. `"\"foo\""`). So basically: `pipe(json: "foo")` would serialize "foo" to JSON, and `pipe(rawJson: "\"foo\"")` would skip the serialization step and assume it's already serialized.